test/pools/PoolCollection.ts

```diff
@@ -1028,6 +1028,613 @@ describe('PoolCollection', () => {
         }
     });
 
+    describe('update trading liquidity', () => {
+        const testDeposit = (tokenData: TokenData) => {
+            let networkSettings: NetworkSettings;
+            let network: TestBancorNetwork;
+            let bnt: IERC20;
+            let bntPool: BNTPool;
+            let masterVault: MasterVault;
+            let poolCollection: TestPoolCollection;
+            let token: TokenWithAddress;
+
+            let provider: SignerWithAddress;
+
+            before(async () => {
+                [, provider] = await ethers.getSigners();
+            });
+
+            beforeEach(async () => {
+                ({ network, bnt, networkSettings, bntPool, masterVault, poolCollection } = await createSystem());
+
+                await networkSettings.setMinLiquidityForTrading(MIN_LIQUIDITY_FOR_TRADING);
+
+                token = await createToken(tokenData);
+            });
+
+            it('should revert when a non-owner attempts to update the trading liquidity', async () => {
+                await expect(
+                    poolCollection
+                        .connect(nonOwner)
+                        .updateTradingLiquidity(token.address)
+                ).to.be.revertedWithError('AccessDenied');
+            });
+
+            it('should revert when attempting to update the trading liquidity of an invalid pool', async () => {
+                await expect(
+                    poolCollection.updateTradingLiquidity(
+                        poolCollection.address
+                    )
+                ).to.be.revertedWithError('DoesNotExist');
+            });
+
+            it('should revert when attempting to update the trading liquidity of a non-existing pool', async () => {
+                await expect(
+                    poolCollection.updateTradingLiquidity(
+                        token.address
+                    )
+                ).to.be.revertedWithError('DoesNotExist');
+            });
+
+            context('with a registered pool', () => {
+                let poolToken: PoolToken;
+
+                const COUNT = 3;
+
+                beforeEach(async () => {
+                    poolToken = await createPool(token, network, networkSettings, poolCollection);
+
+                    await networkSettings.setFundingLimit(token.address, MAX_UINT256);
+                    await poolCollection.setBlockNumber(await latestBlockNumber());
+                });
+
+                enum TradingLiquidityState {
+                    Reset = 0,
+                    Ignore = 1,
+                    Update = 2,
+                    InvalidState = 3
+                }
```

Comment:
This enum (and some of the code below) can be extracted to the global scope and shared between the different cases